src/main/webapp/js/components/entities/users/user-edit.jsx

```diff
@@ -0,0 +1,152 @@
+    super(props);
+
+    this.state = {
+      openConfirmModal: false,
+    };
+  }
+
+  onSubmit = (values) => {
+    const { entityType, fieldConfig } = this.props;
+    const mappedFieldConfig = _.keyBy(_.values(fieldConfig), 'name');
+    const entityValue = {
+      ...values,
+      extraFields: _.map(
+        values.extraFields,
+        (extraField, fieldName) => ({
+          ...extraField,
+          name: fieldName,
+          fieldType: mappedFieldConfig[fieldName].fieldType,
+        }),
+      ),
+    };
+    this.props.updateEntity(entityType, entityValue, () => {
+      Alert.success('User has been updated!', {
+        timeout: ALERT_TIMEOUT,
+      });
+      this.props.history.push('/viewEntity/user');
+    });
+  };
+
+  validate = values => validate(this.props.fieldConfig)(values);
+
+  openConfirmModal = () => {
+    this.setState({ openConfirmModal: true });
+  };
+
+  hideConfirmModal = () => {
+    this.setState({ openConfirmModal: false });
+  };
+
+  deleteEntity = () => {
+    const { entityType, entityId } = this.props;
+    this.props.deleteEntity(entityType, entityId, () => {
+      Alert.success('User has been deleted!', {
+        timeout: ALERT_TIMEOUT,
+      });
+      this.props.history.push('/viewEntity/user');
+    });
+  };
+
+  isAuthorizated = (requiredPermissions) => {
+    const { permissions } = this.props;
+    if (_.isEmpty(requiredPermissions)) {
+      return true;
+    }
+    return _.every(requiredPermissions, permission => _.includes(permissions, permission));
+  };
+
+  render() {
+    const { openConfirmModal } = this.state;
+    const {
+      entityToEdit, isOnline, fieldConfig,
+    } = this.props;
+    return (
+      <div className="container">
+        <div>
+          <h1>
+            {'Edit User'}
+          </h1>
+        </div>
+        <div>
+          <Form
+            onSubmit={this.onSubmit}
+            validate={this.validate}
+            initialValues={entityToEdit}
+            render={({ handleSubmit }) => (
+              <form onSubmit={handleSubmit} className="modal-fields">
+                <div>
+                  {_.map(fieldConfig, elem => renderFormField({ ...elem, name: (elem.base ? elem.name : `extraFields.${elem.name}.value`), editable: this.isAuthorizated([ADMIN]) && elem.name === 'roleIds' ? true : elem.editable }))}
+                </div>
+                <div className="text-center">
+                  <button
+                    type="submit"
+                    className="btn btn-success btn-lg margin-top-sm padding-left-lg padding-right-lg margin-right-sm"
+                    disabled={!isOnline}
+                  >
+                    Update
+                  </button>
+                  <button
+                    type="button"
+                    className="btn btn-danger btn-lg margin-top-sm padding-left-lg padding-right-lg"
+                    onClick={this.openConfirmModal}
+                    disabled={!isOnline}
```

Comment:
you should add a new property for the EntityEdit component to allow disabling the delete button, it can be named 'disableDelete', it should not be required and should be false by default